src/api/v1/endpoints/skill.py

```
@@ -49,6 +51,11 @@ async def create_skill_routine(
                 status_code=500, detail="Agent failed to produce a final response."
             )
 
+    except asyncio.TimeoutError:
+        logger.error("Skill routine generation timed out after %ds", SKILL_TIMEOUT_SECONDS)
+        raise HTTPException(
+            status_code=504, detail="Skill routine generation timed out. Please try again."
+        )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)